django/contrib/postgres/fields/serial.py

```diff
@@ -0,0 +1,67 @@
+from django.db import models
+from django.db.models import NOT_PROVIDED
+from django.db.models.expressions import DatabaseDefault
+from django.utils.translation import gettext_lazy as _
+
+__all__ = ("BigSerialField", "SmallSerialField", "SerialField")
+
+
+class SerialFieldMixin:
+    db_returning = True
+
+    def __init__(self, *args, **kwargs):
+        default = DatabaseDefault()
+
+        if not kwargs.setdefault("blank", True):
+            raise ValueError(f"{self.__class__.__name__} must be blank.")
+        if kwargs.setdefault("null", False):
+            raise ValueError(f"{self.__class__.__name__} must not be null.")
+        if kwargs.setdefault("default", default) is not default:
+            raise ValueError(f"{self.__class__.__name__} cannot have a default.")
+        if kwargs.get("db_default", NOT_PROVIDED) is not NOT_PROVIDED:
+            raise ValueError(
+                f"{self.__class__.__name__} cannot have a database default."
+            )
+
+        super().__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        del kwargs["default"]
+        return name, path, args, kwargs
+
+    def get_prep_value(self, value):
+        value = super().get_prep_value(value)
+        if value is None:
+            value = self.get_default()
+        return value
```

Comment:
It's to allow `SerialModel.objects.create(serial=None)`. If we want to disable this, it can be removed.

The reason it's not in `AutoFieldMixin` is because primary keys are a special case, if it's set to `None`, it's omitted from the `INSERT` query.